src/Evolution/Executables/GrMhd/GhValenciaDivClean/EvolveGhValenciaDivCleanWithHorizon.hpp

```diff
@@ -0,0 +1,185 @@
+        gr::Tags::SpatialMetricCompute<volume_dim, domain_frame, DataVector>,
+        ah::Tags::InverseSpatialMetricCompute<volume_dim, domain_frame>,
+        ah::Tags::ExtrinsicCurvatureCompute<volume_dim, domain_frame>,
+        ah::Tags::SpatialChristoffelSecondKindCompute<volume_dim,
+                                                      domain_frame>>;
+    using vars_to_interpolate_to_target = tmpl::list<
+        gr::Tags::SpatialMetric<volume_dim, domain_frame, DataVector>,
+        gr::Tags::InverseSpatialMetric<volume_dim, domain_frame>,
+        gr::Tags::ExtrinsicCurvature<volume_dim, domain_frame>,
+        gr::Tags::SpatialChristoffelSecondKind<volume_dim, domain_frame>>;
+    using compute_items_on_target = tmpl::append<
+        tmpl::list<StrahlkorperGr::Tags::AreaElementCompute<domain_frame>>,
+        tags_to_observe>;
+    using compute_target_points =
+        intrp::TargetPoints::ApparentHorizon<AhA, ::Frame::Inertial>;
+    using post_interpolation_callback =
+        intrp::callbacks::FindApparentHorizon<AhA, ::Frame::Inertial>;
+    using horizon_find_failure_callback =
+        intrp::callbacks::ErrorOnFailedApparentHorizon;
+    using post_horizon_find_callback =
+        intrp::callbacks::ObserveTimeSeriesOnSurface<tags_to_observe, AhA, AhA>;
+  };
+
+  using interpolation_target_tags = tmpl::list<AhA>;
+  using interpolator_source_vars =
+      tmpl::list<gr::Tags::SpacetimeMetric<volume_dim, domain_frame>,
+                 GeneralizedHarmonic::Tags::Pi<volume_dim, domain_frame>,
+                 GeneralizedHarmonic::Tags::Phi<volume_dim, domain_frame>>;
+
+  using observe_fields = typename GhValenciaDivCleanTemplateBase<
+      EvolutionMetavars>::observe_fields;
+
+  struct factory_creation
+      : tt::ConformsTo<Options::protocols::FactoryCreation> {
+    using factory_classes = tmpl::map<
+        tmpl::pair<
+            Event,
+            tmpl::flatten<tmpl::list<
+                Events::Completion,
+                dg::Events::field_observations<volume_dim, Tags::Time,
+                                               observe_fields,
+                                               analytic_solution_fields>,
+                Events::time_events<system>,
+                intrp::Events::Interpolate<3, AhA, interpolator_source_vars>,
+                intrp::Events::Interpolate<3, InterpolationTargetTags,
+                                           interpolator_source_vars>...>>>,
+        tmpl::pair<StepChooser<StepChooserUse::LtsStep>,
+                   StepChoosers::standard_step_choosers<system>>,
+        tmpl::pair<
+            StepChooser<StepChooserUse::Slab>,
+            StepChoosers::standard_slab_choosers<system, local_time_stepping>>,
+        tmpl::pair<StepController, StepControllers::standard_step_controllers>,
+        tmpl::pair<Trigger, tmpl::append<Triggers::logical_triggers,
+                                         Triggers::time_triggers>>>;
+  };
+
+  using phase_changes =
```

Comment:
why do you need these?  type aliases of public base classes should be visible

Reply:
Unfortunately, that doesn't happen for 'dependent' public base class templates, like in the CRTP trick.

Reply:
ah right...